### PR TITLE
 Decreasing required memory

### PR DESCRIPTION
**Pull request by schubi2**
Replacing old Pkg.ResolvableProperties()and Pkg.ResolvabeDependencies() calls by Y2Packager::Resolvable.any? and Y2Packager::Resolvable.find in order to save memory.

### CODE REVIEW

File: src/modules/Packages.rb [hunk 1]
```diff
@@ -170,11 +167,11 @@ def ResetProposalCache
     # @param [String] format string format string to print summaries in
     # @return a list of selected resolvables
     def ListSelected(what, format)
-      selected = Pkg.ResolvableProperties("", what, "")
+      selected = Y2Packager::Resolvable.find(kind: what)
 
-      selected.select! { |r| r["status"] == :selected }
+      selected.select! { |r| r.status == :selected }
```
Comment (lslezak) [inline src/modules/Packages.rb]: Move it to the Resolvable.find call.

File: src/modules/Packages.rb [hunk 2]
```diff
@@ -429,86 +426,9 @@ def CheckDiskSize(init)
     # Checks which products have been selected for removal and modifies
     # the warning messages accordingly.
     #
-    # @param [Yast::ArgRef] ret reference to map MakeProposal->Summary
-    def CheckOldAddOns(ret)
-      products = Pkg.ResolvableProperties("", :product, "")
```
Comment (lslezak) [inline src/modules/Packages.rb]: Just a question: why we are dropping this?

File: src/modules/Packages.rb [hunk 3]
```diff
@@ -1612,12 +1532,12 @@ def AdjustSourcePropertiesAccordingToProduct(src_id)
...
 
       Builtins.y2milestone("Trying to get repository name from products")
-      all_products = Pkg.ResolvableProperties("", :product, "")
+      all_products = Y2Packager::Resolvable.find(kind: :product)
       new_name = nil
       Builtins.foreach(all_products) do |one_product|
         # source ID matches
-        if Ops.get_integer(one_product, "source", -1) == src_id
-          name = one_product["name"] || ""
+        if (one_product.source || -1) == src_id
```
Comment (lslezak) [inline src/modules/Packages.rb]: You can remove these defaults.

File: src/modules/ProductLicense.rb [hunk 4]
```diff
@@ -1702,9 +1704,9 @@ def product_license(id, tmpdir)
       Pkg.SourceLoad
       ::FileUtils.mkdir_p(tmpdir)
 
-      products = Pkg.ResolvableProperties("", :product, "")
-      products.select! { |p| p["source"] == id }
-      product_names = products.map { |p| p["name"] }.uniq
+      products = Y2Packager::Resolvable.find(kind: :product)
+      products.select! { |p| p.source == id }
```
Comment (lslezak) [inline src/modules/ProductLicense.rb]: Move it to the Resolvable.find call.

File: src/modules/SourceManager.rb [hunk 5]
```diff
@@ -580,7 +581,7 @@ def AddUpdateSources(urls)
           ret = Builtins.add(ret, url) # #180820#c26
 
           # is there any patch available?
-          patches = Pkg.ResolvableProperties("", :patch, "")
+          patches = Y2Packager::Resolvable.find(kind: :patch)
```
Comment (lslezak) [inline src/modules/SourceManager.rb]: Just use `Resolvable.any?` here, we just detect if there is any patch available at all.

File: src/modules/Packages.rb (hunk 2 above)
Comment (schubi2) [reply src/modules/Packages.rb]: We have talked together about it some weeks ago :-)
Currently we do not have any workflow for that.

File: src/modules/SourceManager.rb (hunk 5 above)
Comment (schubi2) [reply src/modules/SourceManager.rb]: But we are also logging how many patches are available.

**Pull request merged**